### PR TITLE
zziplib: Add version 0.13.18

### PR DESCRIPTION
**Pull request by AbrilRBS**
Closes https://github.com/conan-io/conan-center-index/issues/25705

### CODE REVIEW

**Comment by jcar87** [comment]
Hook warnings galore! :P

**Review by uilianries — DISMISSED** [review]
LGTM. https://github.com/gdraheim/zziplib/compare/v0.13.72...v0.13.78#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20a

Between 0.13.77..0.13.78 there is no changes, which means, most those new definitions are older. Still, we have no older versions, so it should not be a problem.

**Comment by jcar87** [comment]
Is the outstanding hook warning correct?

**Review by jcar87 — DISMISSED** [review]
hook warnings need addressing

**Comment by uilianries** [comment]
> Is the outstanding hook warning correct?

Nope. It was not able to deal with that library name: https://github.com/conan-io/conan-center-index/pull/25876/checks?check_run_id=32671715312

The addressed hook information can be found here: https://c3i.jfrog.io/artifactory/cci-build-logs/cci/prod/PR-25876/3/package_build_logs/hook_output_zziplib_0_13_78_a702ebdfc849d51f40651cfd8010aecb_4b22fa0f2e3c3bf57410774c934fd2c53ffc357f.json

> Component zziplib::cpp_info library 'zzip-0' is listed in the recipe, but not found installed at self.cpp_info.libdirs.

The same points to the build log https://c3i.jfrog.io/artifactory/cci-build-logs/cci/prod/PR-25876/3/package_build_logs/build_log_zziplib_0_13_78_a702ebdfc849d51f40651cfd8010aecb_4b22fa0f2e3c3bf57410774c934fd2c53ffc357f.success.txt, which proves the opposite:

> -- Installing: /home/conan/workspace/cci_prod_PR-25876/conan-home/p/b/zzipl19a0a2ea36c48/p/lib/libzzip-0.a

**Comment by uilianries** [comment]
@jcar87 The hook `hook_include_path_does_not_exist` is cleared now. Please, review it again.

**Review by uilianries — APPROVED** [review]

**Review by perseoGI — DISMISSED** [review]

File: recipes/zziplib/all/conanfile.py [hunk 1]
```diff
@@ -51,13 +53,25 @@ def layout(self):
     def requirements(self):
         self.requires("zlib/[>=1.2.11 <2]")
 
+    def validate(self):
+        # Old versions do not support llvm >= 15
+        if (Version(self.version) < "0.13.78"
+                and self.settings.compiler in ("clang", "apple-clang")
+                and Version(self.settings.compiler.version) >= "15.0"):
```
Comment (jcar87) [inline recipes/zziplib/all/conanfile.py:60]: mm apple clang and clang follow different versioning schemes, even if they seem similar - apple clang 15.0 is actually based on llvm 16 - and if the failure is bsed on llvm >-1 there are certain versions of apple-clang 14 that also fall in that category - do we know which one it is?

File: recipes/zziplib/all/conanfile.py [hunk 2]
```diff
@@ -51,13 +53,25 @@ def layout(self):
...
+            raise ConanInvalidConfiguration(f"{self.ref} does not support {self.settings.compiler} >= 15 due to incompatible pointer to integer conversion errors")
```
Comment (jcar87) [inline recipes/zziplib/all/conanfile.py:61]: I think we can leave the `{self.ref}` out? if I remember correctly it is reported by conan before printing it, and any programmatic access is tied to the very reference, so it is redundant?

**Comment by jcar87** [comment]
Ping for the outstanding review comments! IN particular the clang/apple-clang one, the other one is merely cosmetic

File: recipes/zziplib/all/conanfile.py (hunk 1 above)
Comment (AbrilRBS) [reply recipes/zziplib/all/conanfile.py:60]: Failing apple-clang 16:

<details>

```
/Users/ruben/.conan2/p/b/zzipl5c72dfc684354/b/src/zzip/mmapped.c:664:11: error: incompatible pointer to integer conversion initializing 'off_t' (aka 'long long') with an expression of type 'zzip_byte_t *' (aka 'unsigned char *') [-Wint-conversion]
  664 |     off_t offset = zzip_file_header_to_data(header);
      |           ^        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ruben/.conan2/p/b/zzipl5c72dfc684354/b/src/zzip/mmapped.c:666:34: warning: incompatible pointer types initializing 'struct zzip_extra_zip64 *' with an expression of type 'char *' [-Wincompatible-pointer-types]
  666 |         struct zzip_extra_zip64* zip64 =
      |                                  ^
  667 |            zzip_file_header_to_extras(header);
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ruben/.conan2/p/b/zzipl5c72dfc684354/b/src/zzip/mmapped.c:673:34: warning: incompatible pointer types initializing 'struct zzip_extra_zip64 *' with an expression of type 'char *' [-Wincompatible-pointer-types]
  673 |         struct zzip_extra_zip64* zip64 =
      |                                  ^
  674 |            zzip_file_header_to_extras(header);
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ruben/.conan2/p/b/zzipl5c72dfc684354/b/src/zzip/mmapped.c:685:24: error: incompatible integer to pointer conversion assigning to 'Bytef *' (aka 'unsigned char *') from 'off_t' (aka 'long long') [-Wint-conversion]
  685 |     file->zlib.next_in = offset;
```

</details>

failing apple-clang 15:

<details>

```
/Users/ruben/.conan2/p/b/zzipl80b6ead678af1/b/src/zzip/mmapped.c:664:11: error: incompatible pointer to integer conversion initializing 'off_t' (aka 'long long') with an expression of type 'zzip_byte_t *' (aka 'unsigned char *') [-Wint-conversion]
    off_t offset = zzip_file_header_to_data(header);
          ^        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ruben/.conan2/p/b/zzipl80b6ead678af1/b/src/zzip/mmapped.c:666:34: warning: incompatible pointer types initializing 'struct zzip_extra_zip64 *' with an expression of type 'char *' [-Wincompatible-pointer-types]
        struct zzip_extra_zip64* zip64 =
                                 ^
/Users/ruben/.conan2/p/b/zzipl80b6ead678af1/b/src/zzip/mmapped.c:673:34: warning: incompatible pointer types initializing 'struct zzip_extra_zip64 *' with an expression of type 'char *' [-Wincompatible-pointer-types]
        struct zzip_extra_zip64* zip64 =
                                 ^
/Users/ruben/.conan2/p/b/zzipl80b6ead678af1/b/src/zzip/mmapped.c:685:24: error: incompatible integer to pointer conversion assigning to 'Bytef *' (aka 'unsigned char *') from 'off_t' (aka 'long long') [-Wint-conversion]
    file->zlib.next_in = offset;
                       ^ ~~~~~~
```

</details>

Succesful apple-clang 14 compilation:

<details>

```
$ sudo xcode-select -s /Users/ruben/Downloads/Xcode_14.app/Contents/Developer/
$ set -x SDKROOT /Users/ruben/Downloads/Xcode_14.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk/
$ conan create . --version=0.13.72 -b=missing -s:a="compiler.version=14"
WARN: Workspace found: /Users/ruben/coding/conan-center-index
WARN: Workspace ignored as CONAN_WORKSPACE_ENABLE is not set

======== Exporting recipe to the cache ========
zziplib/0.13.72: Exporting package recipe: /Users/ruben/coding/conan-center-index/recipes/zziplib/all/conanfile.py
zziplib/0.13.72: exports: File 'conandata.yml' found. Exporting it...
zziplib/0.13.72: Calling export_sources()
zziplib/0.13.72: Copied 1 '.py' file: conanfile.py
zziplib/0.13.72: Copied 1 '.yml' file: conandata.yml
zziplib/0.13.72: Copied 1 '.patch' file: 0.13.72-001_fix.patch
zziplib/0.13.72: Exported to cache folder: /Users/ruben/.conan2/p/zzipl2b2ce734acae1/e
zziplib/0.13.72: Exported: zziplib/0.13.72#cee9c44071f5fe9d955739080462a230 (2024-12-04 18:58:54 UTC)

======== Input profiles ========
Profile host:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=gnu17
compiler.libcxx=libc++
compiler.version=14
os=Macos
[platform_tool_requires]
cmake/3.30.3
meson/1.5.2
[replace_tool_requires]
meson/*: meson/[>=1.0 <2]
[conf]


Profile build:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=gnu17
compiler.libcxx=libc++
compiler.version=14
os=Macos
[platform_tool_requires]
cmake/3.30.3
meson/1.5.2
[replace_tool_requires]
meson/*: meson/[>=1.0 <2]
[conf]



======== Computing dependency graph ========
Graph root
    cli
Requirements
    zlib/1.3.1#f52e03ae3d251dec704634230cd806a2 - Cache
    zziplib/0.13.72#cee9c44071f5fe9d955739080462a230 - Cache
Resolved version ranges
    zlib/[>=1.2.11 <2]: zlib/1.3.1

======== Computing necessary packages ========
WARN: Workspace found: /Users/ruben/coding/conan-center-index
WARN: Workspace ignored as CONAN_WORKSPACE_ENABLE is not set
Requirements
    zlib/1.3.1#f52e03ae3d251dec704634230cd806a2:76f7d863f21b130b4e6527af3b1d430f7f8edbea#754b8c19fa166622829fec5dbbf8095f - Cache
    zziplib/0.13.72#cee9c44071f5fe9d955739080462a230:cdb581cd411fbaf6ea6beeabd9c068f6f6595dfe - Build

======== Installing packages ========
zlib/1.3.1: Already installed! (1 of 2)

-------- Installing package zziplib/0.13.72 (2 of 2) --------
zziplib/0.13.72: Building from source
zziplib/0.13.72: Package zziplib/0.13.72:cdb581cd411fbaf6ea6beeabd9c068f6f6595dfe
zziplib/0.13.72: Copying sources to build folder
zziplib/0.13.72: Building your package in /Users/ruben/.conan2/p/b/zzipl560ce7170337e/b
zziplib/0.13.72: Calling generate()
zziplib/0.13.72: Generators folder: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/b/build/Release/generators
zziplib/0.13.72: CMakeToolchain generated: conan_toolchain.cmake
zziplib/0.13.72: CMakeToolchain generated: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/b/build/Release/generators/CMakePresets.json
zziplib/0.13.72: CMakeToolchain generated: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/b/src/CMakeUserPresets.json
zziplib/0.13.72: CMakeDeps necessary find_package() and targets for your CMakeLists.txt
    find_package(ZLIB)
    target_link_libraries(... ZLIB::ZLIB)
zziplib/0.13.72: Generating aggregated env files
zziplib/0.13.72: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
zziplib/0.13.72: Calling build()
zziplib/0.13.72: Apply patch (conan): fix install path
zziplib/0.13.72: Running CMake.configure()
zziplib/0.13.72: RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/Users/ruben/.conan2/p/b/zzipl560ce7170337e/p" -DCMAKE_POLICY_DEFAULT_CMP0077="NEW" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/Users/ruben/.conan2/p/b/zzipl560ce7170337e/b/src"
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- Using Conan toolchain: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/b/build/Release/generators/conan_toolchain.cmake
-- Conan toolchain: Setting CMAKE_POSITION_INDEPENDENT_CODE=ON (options.fPIC)
-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
-- The C compiler identification is AppleClang 14.0.3.14030022
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Users/ruben/Downloads/Xcode_14.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
CMake Deprecation Warning at zzip/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- Found UnixCommands: /bin/bash
-- Looking for include file byteswap.h
-- Looking for include file byteswap.h - not found
-- Looking for include file direct.h
-- Looking for include file direct.h - not found
-- Looking for include file dirent.h
-- Looking for include file dirent.h - found
-- Looking for include file dlfcn.h
-- Looking for include file dlfcn.h - found
-- Looking for include file fnmatch.h
-- Looking for include file fnmatch.h - found
-- Looking for include file inttypes.h
-- Looking for include file inttypes.h - found
-- Looking for include file io.h
-- Looking for include file io.h - not found
-- Looking for include file memory.h
-- Looking for include file memory.h - found
-- Looking for include file ndir.h
-- Looking for include file ndir.h - not found
-- Looking for include file stdint.h
-- Looking for include file stdint.h - found
-- Looking for include file stdlib.h
-- Looking for include file stdlib.h - found
-- Looking for strcasecmp
-- Looking for strcasecmp - found
-- Looking for include file strings.h
-- Looking for include file strings.h - found
-- Looking for include file string.h
-- Looking for include file string.h - found
-- Looking for strndup
-- Looking for strndup - found
-- Looking for include file sys/dir.h
-- Looking for include file sys/dir.h - found
-- Looking for include file sys/int_types.h
-- Looking for include file sys/int_types.h - not found
-- Looking for include file sys/mman.h
-- Looking for include file sys/mman.h - found
-- Looking for include file sys/ndir.h
-- Looking for include file sys/ndir.h - not found
-- Looking for include file sys/param.h
-- Looking for include file sys/param.h - found
-- Looking for include file sys/stat.h
-- Looking for include file sys/stat.h - found
-- Looking for include file sys/types.h
-- Looking for include file sys/types.h - found
-- Looking for include file unistd.h
-- Looking for include file unistd.h - found
-- Looking for pread
-- Looking for pread - found
-- Looking for include file winbase.h
-- Looking for include file winbase.h - not found
-- Looking for include file windows.h
-- Looking for include file windows.h - not found
-- Looking for include file winnt.h
-- Looking for include file winnt.h - not found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of int
-- Check size of int - done
-- Check size of long
-- Check size of long - done
-- Check size of short
-- Check size of short - done
-- Looking for 4 include files stdlib.h, ..., float.h
-- Looking for 4 include files stdlib.h, ..., float.h - found
-- Check size of off_t_32
-- Check size of off_t_32 - done
-- Check size of off_t_64
-- Check size of off_t_64 - done
-- Check size of off64_t
-- Check size of off64_t - failed
-- found sizeof(off_t/32)=8 and sizeof(off_t/64)=8
-- Conan: Target declared 'ZLIB::ZLIB'
-- lib zzipfseeko  to be compiled: True
-- lib zzipmmapped to be compiled: True
-- lib zzip symlinks as by libtool: ON
-- installing zzip compat headers: ON
-- installing zzip pkgconfig files: ON
CMake Deprecation Warning at zzipwrap/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- Found PkgConfig: /opt/homebrew/bin/pkg-config (found version "2.3.0")
-- Conan: Target declared 'ZLIB::ZLIB'
-- Checking for one of the modules 'zzip'
-- lib zzipwrap symlinks as by libtool: ON
-- installing zzipwrap pkgconfig files: ON
-- Configuring done (8.8s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/b/build/Release

zziplib/0.13.72: Running CMake.build()
zziplib/0.13.72: RUN: cmake --build "/Users/ruben/.conan2/p/b/zzipl560ce7170337e/b/build/Release" -- -j12
[  4%] Building C object zzip/CMakeFiles/libzzip.dir/dir.c.o
[  8%] Building C object zzip/CMakeFiles/libzzip.dir/err.c.o
[ 12%] Building C object zzip/CMakeFiles/libzzip.dir/info.c.o
[ 16%] Building C object zzip/CMakeFiles/libzzipfseeko.dir/fseeko.c.o
[ 20%] Building C object zzip/CMakeFiles/libzzip.dir/plugin.c.o
[ 25%] Building C object zzip/CMakeFiles/libzzip.dir/file.c.o
[ 29%] Building C object zzip/CMakeFiles/libzzip.dir/write.c.o
[ 33%] Building C object zzip/CMakeFiles/libzzipmmapped.dir/mmapped.c.o
[ 37%] Generating zziplib.pc
[ 41%] Building C object zzip/CMakeFiles/libzzip.dir/stat.c.o
[ 45%] Generating zzipwrap.pc
[ 50%] Generating compat/zzip.h
[ 54%] Building C object zzip/CMakeFiles/libzzip.dir/zip.c.o
[ 58%] Building C object zzip/CMakeFiles/libzzipfseeko.dir/fetch.c.o
[ 62%] Building C object zzip/CMakeFiles/libzzip.dir/fetch.c.o
[ 66%] Building C object zzip/CMakeFiles/libzzipmmapped.dir/memdisk.c.o
/Users/ruben/.conan2/p/b/zzipl560ce7170337e/b/src/zzip/mmapped.c:664:11: warning: incompatible pointer to integer conversion initializing 'off_t' (aka 'long long') with an expression of type 'zzip_byte_t *' (aka 'unsigned char *') [-Wint-conversion]
    off_t offset = zzip_file_header_to_data(header);
          ^        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ruben/.conan2/p/b/zzipl560ce7170337e/b/src/zzip/mmapped.c:666:34: warning: incompatible pointer types initializing 'struct zzip_extra_zip64 *' with an expression of type 'char *' [-Wincompatible-pointer-types]
        struct zzip_extra_zip64* zip64 =
                                 ^
/Users/ruben/.conan2/p/b/zzipl560ce7170337e/b/src/zzip/mmapped.c:673:34: warning: incompatible pointer types initializing 'struct zzip_extra_zip64 *' with an expression of type 'char *' [-Wincompatible-pointer-types]
        struct zzip_extra_zip64* zip64 =
                                 ^
/Users/ruben/.conan2/p/b/zzipl560ce7170337e/b/src/zzip/mmapped.c:685:24: warning: incompatible integer to pointer conversion assigning to 'Bytef *' (aka 'unsigned char *') from 'off_t' (aka 'long long') [-Wint-conversion]
    file->zlib.next_in = offset;
                       ^ ~~~~~~
[ 70%] Building C object zzip/CMakeFiles/libzzipmmapped.dir/fetch.c.o
4 warnings generated.
[ 75%] Linking C static library libzzipfseeko-0.a
/Users/ruben/.conan2/p/b/zzipl560ce7170337e/b/src/zzip/memdisk.c:239:37: warning: initializing 'zzip_byte_t *' (aka 'unsigned char *') with an expression of type 'char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
        zzip_byte_t *_zzip_restrict ext1_ptr = zzip_disk_entry_to_extras(entry);
                                    ^          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/ruben/.conan2/p/b/zzipl560ce7170337e/b/src/zzip/memdisk.c:241:37: warning: initializing 'zzip_byte_t *' (aka 'unsigned char *') with an expression of type 'char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
        zzip_byte_t *_zzip_restrict ext2_ptr = zzip_file_header_to_extras(header);
                                    ^          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[ 79%] Linking C static library libzzip-0.a
2 warnings generated.
[ 83%] Linking C static library libzzipmmapped-0.a
[ 83%] Built target pkgconfigs
[ 83%] Built target libzzipfseeko
[ 83%] Built target libzzip
[ 83%] Built target libzzipmmapped
[ 83%] Built target compat
[ 83%] Built target pkgconfig
[ 87%] Building C object zzipwrap/CMakeFiles/libzzipwrap.dir/wrap.c.o
[ 91%] Linking C static library libzzipwrap-0.a
[ 91%] Built target libzzipwrap
[ 95%] Building C object zzipwrap/CMakeFiles/zzipwrap.dir/zzipwrap.c.o
[100%] Linking C executable zzipwrap
[100%] Built target zzipwrap

zziplib/0.13.72: Package 'cdb581cd411fbaf6ea6beeabd9c068f6f6595dfe' built
zziplib/0.13.72: Build folder /Users/ruben/.conan2/p/b/zzipl560ce7170337e/b/build/Release
zziplib/0.13.72: Generating the package
zziplib/0.13.72: Packaging in folder /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p
zziplib/0.13.72: Calling package()
zziplib/0.13.72: Running CMake.install()
zziplib/0.13.72: RUN: cmake --install "/Users/ruben/.conan2/p/b/zzipl560ce7170337e/b/build/Release" --prefix "/Users/ruben/.conan2/p/b/zzipl560ce7170337e/p"
-- Install configuration: "Release"
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/lib/pkgconfig/zziplib.pc
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/lib/pkgconfig/zzipmmapped.pc
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/lib/pkgconfig/zzipfseeko.pc
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zzip/lib.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zzip/zzip.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zzip/format.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zzip/types.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zzip/conf.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zzip/_msvc.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zzip/file.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zzip/info.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zzip/plugin.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zzip/write.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zzip/fetch.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zzip/stdint.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zzip/zzip32.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zzip/autoconf.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zzip/_config.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/lib/libzzip-0.a
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zzip/fseeko.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/lib/libzzipfseeko-0.a
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zzip/mmapped.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zzip/memdisk.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/lib/libzzipmmapped-0.a
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zzip.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zziplib.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zzip-io.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/lib/pkgconfig/zzipwrap.pc
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/include/zzip/wrap.h
-- Installing: /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p/lib/libzzipwrap-0.a

zziplib/0.13.72: [HOOK - hook_ms_runtime_files.py] post_package(): Skipping zziplib from checking MS files: Not msvc.
zziplib/0.13.72: package(): Packaged 1 '.LIB' file: COPYING.LIB
zziplib/0.13.72: package(): Packaged 22 '.h' files
zziplib/0.13.72: package(): Packaged 4 '.a' files: libzzipfseeko-0.a, libzzip-0.a, libzzipwrap-0.a, libzzipmmapped-0.a
zziplib/0.13.72: Created package revision 65826c88c77545ec031a6846f41ea256
zziplib/0.13.72: Package 'cdb581cd411fbaf6ea6beeabd9c068f6f6595dfe' created
zziplib/0.13.72: Full package reference: zziplib/0.13.72#cee9c44071f5fe9d955739080462a230:cdb581cd411fbaf6ea6beeabd9c068f6f6595dfe#65826c88c77545ec031a6846f41ea256
zziplib/0.13.72: Package folder /Users/ruben/.conan2/p/b/zzipl560ce7170337e/p
WARN: deprecated: Usage of deprecated Conan 1.X features that will be removed in Conan 2.X:
WARN: deprecated:     'cpp_info.names' used in: zlib/1.3.1

======== Launching test_package ========

======== Computing dependency graph ========
Graph root
    zziplib/0.13.72 (test package): /Users/ruben/coding/conan-center-index/recipes/zziplib/all/test_package/conanfile.py
Requirements
    zlib/1.3.1#f52e03ae3d251dec704634230cd806a2 - Cache
    zziplib/0.13.72#cee9c44071f5fe9d955739080462a230 - Cache

======== Computing necessary packages ========
Requirements
    zlib/1.3.1#f52e03ae3d251dec704634230cd806a2:76f7d863f21b130b4e6527af3b1d430f7f8edbea#754b8c19fa166622829fec5dbbf8095f - Cache
    zziplib/0.13.72#cee9c44071f5fe9d955739080462a230:cdb581cd411fbaf6ea6beeabd9c068f6f6595dfe#65826c88c77545ec031a6846f41ea256 - Cache

======== Installing packages ========
zlib/1.3.1: Already installed! (1 of 2)
zziplib/0.13.72: Already installed! (2 of 2)
WARN: deprecated: Usage of deprecated Conan 1.X features that will be removed in Conan 2.X:
WARN: deprecated:     'cpp_info.names' used in: zlib/1.3.1

======== Testing the package ========
Removing previously existing 'test_package' build folder: /Users/ruben/coding/conan-center-index/recipes/zziplib/all/test_package/build/apple-clang-14-armv8-gnu17-release
zziplib/0.13.72 (test package): Test package build: build/apple-clang-14-armv8-gnu17-release
zziplib/0.13.72 (test package): Test package build folder: /Users/ruben/coding/conan-center-index/recipes/zziplib/all/test_package/build/apple-clang-14-armv8-gnu17-release
zziplib/0.13.72 (test package): Writing generators to /Users/ruben/coding/conan-center-index/recipes/zziplib/all/test_package/build/apple-clang-14-armv8-gnu17-release/generators
zziplib/0.13.72 (test package): Generator 'CMakeToolchain' calling 'generate()'
zziplib/0.13.72 (test package): CMakeToolchain generated: conan_toolchain.cmake
zziplib/0.13.72 (test package): CMakeToolchain generated: /Users/ruben/coding/conan-center-index/recipes/zziplib/all/test_package/build/apple-clang-14-armv8-gnu17-release/generators/CMakePresets.json
zziplib/0.13.72 (test package): CMakeToolchain generated: /Users/ruben/coding/conan-center-index/recipes/zziplib/all/test_package/CMakeUserPresets.json
zziplib/0.13.72 (test package): Generator 'CMakeDeps' calling 'generate()'
zziplib/0.13.72 (test package): CMakeDeps necessary find_package() and targets for your CMakeLists.txt
    find_package(zziplib)
    target_link_libraries(... zziplib::zziplib)
zziplib/0.13.72 (test package): Generator 'VirtualRunEnv' calling 'generate()'
zziplib/0.13.72 (test package): Generating aggregated env files
zziplib/0.13.72 (test package): Generated aggregated env files: ['conanrun.sh', 'conanbuild.sh']

======== Testing the package: Building ========
zziplib/0.13.72 (test package): Calling build()
zziplib/0.13.72 (test package): Running CMake.configure()
zziplib/0.13.72 (test package): RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/Users/ruben/coding/conan-center-index/recipes/zziplib/all/test_package" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/Users/ruben/coding/conan-center-index/recipes/zziplib/all/test_package"
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- Using Conan toolchain: /Users/ruben/coding/conan-center-index/recipes/zziplib/all/test_package/build/apple-clang-14-armv8-gnu17-release/generators/conan_toolchain.cmake
-- Conan toolchain: Defining libcxx as C++ flags: -stdlib=libc++
-- Conan toolchain: C++ Standard 17 with extensions ON
-- The C compiler identification is AppleClang 14.0.3.14030022
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Users/ruben/Downloads/Xcode_14.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Conan: Component target declared 'zziplib::zzip'
-- Conan: Component target declared 'zziplib::zzipmmapped'
-- Conan: Component target declared 'zziplib::zzipfseeko'
-- Conan: Component target declared 'zziplib::zzipwrap'
-- Conan: Target declared 'zziplib::zziplib'
-- Conan: Target declared 'ZLIB::ZLIB'
-- Configuring done (0.4s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/ruben/coding/conan-center-index/recipes/zziplib/all/test_package/build/apple-clang-14-armv8-gnu17-release

zziplib/0.13.72 (test package): Running CMake.build()
zziplib/0.13.72 (test package): RUN: cmake --build "/Users/ruben/coding/conan-center-index/recipes/zziplib/all/test_package/build/apple-clang-14-armv8-gnu17-release" -- -j12
[ 50%] Building C object CMakeFiles/test_package.dir/test_package.c.o
[100%] Linking C executable test_package
[100%] Built target test_package


======== Testing the package: Executing test ========
zziplib/0.13.72 (test package): Running test()
zziplib/0.13.72 (test package): RUN: ./test_package
zzip code 0 means Undefined error: 0

```

</details>

Comment (AbrilRBS) [reply recipes/zziplib/all/conanfile.py:60]: So in summary, it seems like this check is actually ok

**Review by AbrilRBS — COMMENTED** [review]
Provided logs to show that the llvm/apple-clang checks are correct, I think this can be merged as is

cf @jcar87

**Review by perseoGI — APPROVED** [review]
Looking good

File: recipes/zziplib/all/conanfile.py (hunk 1 above)
Comment (jcar87) [reply recipes/zziplib/all/conanfile.py:60]: Awesome! thanks for testing

**Review by jcar87 — APPROVED** [review]

**Pull request merged**